server/app/controllers/UserController.ts

```diff
@@ -63,13 +63,50 @@ export class UserController extends BaseHttpController {
         const accessToken = generateToken(user);
         response.jwt = accessToken;
       }
-
       return this.json(response);
     } catch (err) {
       return this.json(err.message, 400);
     }
   }
 
+  @Doc('Forgot password')
+  /**
+   * @desc        Send email to existing user
+   * @route       POST /user/forgot
+   * @access      Public
+   * @param req
+   * @returns     User JSON Format + Jwt token
+   */
+  @httpPost('/forgot')
+  public async forgot(req: Request): Promise<results.JsonResult> {
+    try {
+      const promise = await this.userService.forgotPassword(req.body.url, req.body);
+      return this.json(promise, 200);
+    } catch (err) {
+      return this.json(err.message, 300);
+    }
+  }
+
+  @Doc('Reset password')
+  /**
+   * @desc        Reset password with token
+   * @route       POST /user/reset
+   * @access      Public
+   * @param req
+   * @returns     User JSON Format + Jwt token
+   */
+  @httpPost('/reset')
+  public async reset(req: Request): Promise<results.JsonResult> {
+    try {
+      const token = req.body.token;
+      const pass = req.body.pass;
```

Comment:
just for safe measure I would add a safety check here in case the request is missing a token or  pass

Reply:
I change it to [req.body?.token], is that good? Or there is other safety check beside that

Reply:
@Ming424 I think you should pass the request body as it is to the service methods (`forgotPassword` and `resetPassword`) and let those services check if the necessary body properties are present and throw/handle missing properties if needed.